File: Project Euler/p7.cpp
// 10001st prime
#include <bits/stdc++.h>

using namespace std;

bool checkPrime(long n) {
    if (n <= 1) {
        return false;
    } else {
        for (int i = 2; i < int(sqrt(n)); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

int main() {
    long counter = 0;
    long prime = 2;
    while (counter < 10001) {
        if (checkPrime(prime)) {
            counter ++;
        }
        prime ++;
    }
    cout << prime-1 << endl;
    return 0;
}

File: Project Euler/p7.py
# 10001st prime
import math

def check_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n))):
        if n % i == 0:
            return False
    return True

def main():
    counter = 0
    num = 2
    while counter < 10001:
        if check_prime(num):
            counter += 1
        num += 1
    return num - 1

if __name__ == "__main__":
    print(main())
